src/Session/Session.php

```diff
@@ -102,8 +107,7 @@ public function setData($data)
     public function getUser()
     {
         $user_uuid = $this->get('user_uuid');
-        // TODO: Remove this direct instantiation to make this more testable
-        $user = new User((object)array('id' => $user_uuid));
+        $user = $this->getContainer()->get(User::class, [(object)array('id' => $user_uuid)]);
```

Comment:
"Referencing the container outside of the factory method is nearly always a bug." - Larry Garfield.

I don't think it is necessary to adhere to this principal 100%, as doing so cuts out the ability to use the container for classes that need to dynamically construct objects. However, I often find myself returning to Larry Garfield quotes any time I am tempted to "break a rule", and see if there's a better way to do it.  Here, Larry means constructors and static factory methods when he says "factory", specifically the later, as if you are really strict, you can pass the container into the static factory and never store it in the class. I don't know that we are quite there yet, but if we could grab the user uuid and do the user lookup at the time that the Session object is being constructed, I think that would be an improvement.